packages/vega-scenegraph/src/util/canvas/resize.js

```diff
@@ -4,21 +4,23 @@ function devicePixelRatio() {
 
 var pixelRatio = devicePixelRatio();
 
-export default function(canvas, width, height, origin, scaleFactor, opt) {
+export default function(canvas, width, height, origin, scaleFactor, ctx, opt) {
   const inDOM = typeof HTMLElement !== 'undefined'
               && canvas instanceof HTMLElement
               && canvas.parentNode != null,
-        context = canvas.getContext('2d'),
+        context = ctx || canvas.getContext('2d'),
         ratio = inDOM ? pixelRatio : scaleFactor;
 
-  canvas.width = width * ratio;
-  canvas.height = height * ratio;
+  if (canvas) {
+    canvas.width = width * ratio;
+    canvas.height = height * ratio;
+  }
 
   for (const key in opt) {
```

Comment:
With a passed-in context, I don't think we want to apply all the context-specific operations in the `resize` utility. For example, here we may overwrite desired properties (set externally) of the node-canvas context. That said, the bigger concern is that the transform matrix gets completely overwritten later in the method via `setTransform`!

Reply:
Applying setTransform to `context` is fine as long as the user calls `save` (and later `restore`) on it, right?